controllers/k8ssandra/k8ssandracluster_controller_test.go

```diff
@@ -2398,6 +2405,10 @@ func changeClusterNameFails(t *testing.T, ctx context.Context, f *framework.Fram
 	err = f.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterName}, k8c)
 	require.NoError(err, "failed to get K8ssandraCluster")
 
+	k8c.Spec.Cassandra.Datacenters[0].Meta.Name = "newDC1"
```

Comment:
issue: This breaks the clusterName test, since the error text isn't tested. As you don't reset the status, the next test will fail, even if the clusterName check was removed since the DC name check would be checked.

Reply:
Ah, OK that's true. It will fail because of the DC name change, even if the newClusterName was what caused the failure. Nice catch, thank you.